docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,999 @@
+    "bounds_max = np.percentile(training_data, 95, axis=0)\n",
+    "bounds = []\n",
+    "for i, _ in enumerate(bounds_min):\n",
+    "    bounds.append([bounds_min[i], bounds_max[i]])\n",
+    "\n",
+    "# Determine data resolution for each dimension of the training data in terms\n",
+    "# of the number of qubits used to represent each data dimension.\n",
+    "data_dim = [3, 3]\n",
+    "\n",
+    "# Pre-processing, i.e., discretization of the data (gridding)\n",
+    "(training_data, _, grid_elements, prob_data) = discretize_and_truncate(\n",
+    "    training_data,\n",
+    "    np.asarray(bounds),\n",
+    "    data_dim,\n",
+    "    return_data_grid_elements=True,\n",
+    "    return_prob=True,\n",
+    "    prob_non_zero=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "We convert data arrays into tensors and create a data loader from our training data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "training_data = torch.tensor(training_data, dtype=torch.float)\n",
+    "grid_elements = torch.tensor(grid_elements, dtype=torch.float)\n",
+    "\n",
+    "# Define the training batch size\n",
+    "batch_size = 300\n",
+    "dataloader = DataLoader(training_data, batch_size=batch_size, shuffle=True, drop_last=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Specify Backend\n",
+    "Next, we need to choose a backend that is used to run the quantum generator.\n",
+    "The presented method is compatible with all shot-based backends (qasm, fake hardware, real hardware) provided by Qiskit.\n",
+    "\n",
+    "First, we create a quantum instance for the training where the batch size defines the number of shots."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "backend = Aer.get_backend(\"aer_simulator\")\n",
+    "qi_training = QuantumInstance(backend, shots=batch_size)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Then we create a quantum instance for the evaluation, we choose a higher number of shots to get better insights."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "qi_sampling = QuantumInstance(backend, shots=10000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Initialization of quantum generator and the respective gradient\n",
```

Comment:
I would change this title, it is not initializing yet, just setting up helper functions to initialize later on.

Reply:
Renamed to "Definition of quantum generator", similar to the discriminator section.